src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -106,7 +106,7 @@ private BoundExpression CompareWithConstant(BoundExpression input, BoundExpressi
                 );
         }
 
-        BoundExpression MakeDeclarationPattern(SyntaxNode syntax, BoundExpression input, LocalSymbol target, bool requiresNullTest)
+        BoundExpression MakeDeclarationPattern(SyntaxNode syntax, BoundExpression input, BoundExpression target, bool requiresNullTest)
```

Comment:
> That kind of cross-component coupling makes the code more fragile than it needs to be.

I don't think the assert would make the code more fragile than it is already. On the positive side, it would make explicit the assumptions that you are making and would verify/enforce them. 

---
In reply to: [77746892](https://github.com/dotnet/roslyn/pull/13535#discussion_r77746892) [](ancestors = 77746892,77736440,77735261,77712197)